### ISSUE TITLE
race between List::Util and Math::BigInt::Fastcalc

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8310]
Migrated from [rt.perl.org#38404](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=38404) (status was 'resolved')

Searchable as RT38404$

**Comment by p5pRT** [comment]
### From @nwc10

#### Created by @nwc10



The perl make process still has at least 1 race condition\. I've had this
exact failure before\, but didn't grab the output at the time\.


Writing Makefile for MIME&#8203;::Base64
&nbsp;        Making Math&#8203;::BigInt&#8203;::FastCalc \(dynamic\)
Can't locate Scalar/Util\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./\.\./\.\./\.\./lib /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4/i386\-freebsd /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4 /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4/i386\-freebsd /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4 /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl \.\) at \.\./\.\./\.\./\.\./lib/overload\.pm line 88\.
Warning&#8203;: No Makefile\!
make&#8203;: don't know how to make config\. Stop
make config failed\, continuing anyway\.\.\.
make&#8203;: don't know how to make all\. Stop
\*\*\* Error code 2
Writing compiled form
\.\./\.\./\.\./miniperl "\-I\.\./\.\./\.\./lib" "\-I\.\./\.\./\.\./lib" \.\./\.\./\.\./lib/ExtUtils/xsubpp  \-typemap \.\./\.\./\.\./lib/ExtUtils/typemap  Util\.xs > Util\.xsc && mv Util\.xsc Util\.c
cp lib/List/Util\.pm \.\./\.\./\.\./lib/List/Util\.pm
cp lib/Scalar/Util\.pm \.\./\.\./\.\./lib/Scalar/Util\.pm
Running Mkbootstrap for Encode \(\)
chmod 644 Encode\.bs
cp bin/piconv blib/script/piconv
\.\./\.\./miniperl "\-I\.\./\.\./lib" "\-I\.\./\.\./lib" "\-MExtUtils&#8203;::MY" \-e "MY\->fixin\(shift\)" blib/script/piconv
Running Mkbootstrap for List&#8203;::Util \(\)
Writing compiled form


That's all I have\. It seems that if the Makefile\.PL for Math&#8203;::BigInt&#8203;::FastCalc
is run at \*exactly\* the wrong time\, somewhere in the middle of List&#8203;::Util
being copied into lib\, then it fails\.

But I've been unable to replicate this by doing nasty things like seeding
lib with all generated files truncated to zero length \(so that require will
fail\)

So I'm stuck

Nicholas Clark

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.9.4:

Configured by nicholas at Thu Feb  2 00:04:15 CET 2006.

Summary of my perl5 (revision 5 version 9 subversion 4) configuration:
  Platform:
    osname=freebsd, osvers=6.0-stable, archname=i386-freebsd
    uname='freebsd profane.mongueurs.net 6.0-stable freebsd 6.0-stable #4: wed jan 25 15:42:18 cet 2006 root@profane.mongueurs.net:usrobjusrsrcsysprofane i386 '
    config_args='-Dusedevel=y -Dcc=ccache gcc -Dld=gcc -Ubincompat5005 -Uinstallusrbinperl -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -Doptimize=-Os -Uusethreads -Uuse64bitint -Uusemymalloc -Duseperlio -Dprefix=~/Sandpit/snap5.9.x-27046 -Dinstallman1dir=none -Dinstallman3dir=none -de'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include',
    optimize='-Os',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include'
    ccversion='', gccversion='3.4.4 [FreeBSD] 20050518', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Wl,-E  -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lm -lcrypt -lutil -lc
    perllibs=-lm -lcrypt -lutil -lc
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fPIC', lddlflags='-shared  -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.9.4:
    lib
    /home/nicholas/Sandpit/snap5.9.x-27046/lib/perl5/5.9.4/i386-freebsd
    /home/nicholas/Sandpit/snap5.9.x-27046/lib/perl5/5.9.4
    /home/nicholas/Sandpit/snap5.9.x-27046/lib/perl5/site_perl/5.9.4/i386-freebsd
    /home/nicholas/Sandpit/snap5.9.x-27046/lib/perl5/site_perl/5.9.4
    /home/nicholas/Sandpit/snap5.9.x-27046/lib/perl5/site_perl
    .


Environment for perl 5.9.4:
    HOME=/home/nicholas
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/nicholas/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/usr/X11R6/bin:/home/nicholas/bin:/usr/local/sbin:/sbin:/usr/sbin:.
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @smpeters

On Wed\, Feb 01\, 2006 at 04&#8203;:33&#8203;:08PM \-0800\, Nicholas Clark wrote&#8203;:
> \# New Ticket Created by  Nicholas Clark 
> \# Please include the string&#8203;:  \[perl \#38404\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=38404 >
> 
> 
> 
> This is a bug report for perl from nick@&#8203;ccl4\.org\,
> generated with the help of perlbug 1\.35 running under perl 5\.9\.4\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> The perl make process still has at least 1 race condition\. I've had this
> exact failure before\, but didn't grab the output at the time\.
> 
> 
> Writing Makefile for MIME&#8203;::Base64
>         Making Math&#8203;::BigInt&#8203;::FastCalc \(dynamic\)
> Can't locate Scalar/Util\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./\.\./\.\./\.\./lib /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4/i386\-freebsd /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4 /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4/i386\-freebsd /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4 /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl \.\) at \.\./\.\./\.\./\.\./lib/overload\.pm line 88\.
> Warning&#8203;: No Makefile\!
> make&#8203;: don't know how to make config\. Stop
> make config failed\, continuing anyway\.\.\.
> make&#8203;: don't know how to make all\. Stop
> \*\*\* Error code 2
> Writing compiled form
> \.\./\.\./\.\./miniperl "\-I\.\./\.\./\.\./lib" "\-I\.\./\.\./\.\./lib" \.\./\.\./\.\./lib/ExtUtils/xsubpp  \-typemap \.\./\.\./\.\./lib/ExtUtils/typemap  Util\.xs > Util\.xsc && mv Util\.xsc Util\.c
> cp lib/List/Util\.pm \.\./\.\./\.\./lib/List/Util\.pm
> cp lib/Scalar/Util\.pm \.\./\.\./\.\./lib/Scalar/Util\.pm
> Running Mkbootstrap for Encode \(\)
> chmod 644 Encode\.bs
> cp bin/piconv blib/script/piconv
> \.\./\.\./miniperl "\-I\.\./\.\./lib" "\-I\.\./\.\./lib" "\-MExtUtils&#8203;::MY" \-e "MY\->fixin\(shift\)" blib/script/piconv
> Running Mkbootstrap for List&#8203;::Util \(\)
> Writing compiled form
> 

Actually\, Math&#8203;::BigInt&#8203;::FastCalc is an innocent bystander\.  overload\.pm
C\<require>s Scalar&#8203;::Util\, but several modules built first use it\.  
Encode\, for example\, which was being built in another process\.  Here are
the modules using overload in the ext directory\.

> 
> That's all I have\. It seems that if the Makefile\.PL for Math&#8203;::BigInt&#8203;::FastCalc
> is run at \*exactly\* the wrong time\, somewhere in the middle of List&#8203;::Util
> being copied into lib\, then it fails\.
> 
> But I've been unable to replicate this by doing nasty things like seeding
> lib with all generated files truncated to zero length \(so that require will
> fail\)
> 
> So I'm stuck
> 
> Nicholas Clark
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=low
> \-\-\-
> Site configuration information for perl 5\.9\.4&#8203;:
> 
> Configured by nicholas at Thu Feb  2 00&#8203;:04&#8203;:15 CET 2006\.
> 
> Summary of my perl5 \(revision 5 version 9 subversion 4\) configuration&#8203;:
>   Platform&#8203;:
>     osname=freebsd\, osvers=6\.0\-stable\, archname=i386\-freebsd
>     uname='freebsd profane\.mongueurs\.net 6\.0\-stable freebsd 6\.0\-stable \#4&#8203;: wed jan 25 15&#8203;:42&#8203;:18 cet 2006 root@&#8203;profane\.mongueurs\.net&#8203;:usrobjusrsrcsysprofane i386 '
>     config\_args='\-Dusedevel=y \-Dcc=ccache gcc \-Dld=gcc \-Ubincompat5005 \-Uinstallusrbinperl \-Dcf\_email=nick@&#8203;ccl4\.org \-Dperladmin=nick@&#8203;ccl4\.org \-Dinc\_version\_list=  \-Dinc\_version\_list\_init=0 \-Doptimize=\-Os \-Uusethreads \-Uuse64bitint \-Uusemymalloc \-Duseperlio \-Dprefix=~/Sandpit/snap5\.9\.x\-27046 \-Dinstallman1dir=none \-Dinstallman3dir=none \-de'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     useithreads=undef\, usemultiplicity=undef
>     useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
>     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='ccache gcc'\, ccflags ='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe \-Wdeclaration\-after\-statement \-I/usr/local/include'\,
>     optimize='\-Os'\,
>     cppflags='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe \-Wdeclaration\-after\-statement \-I/usr/local/include'
>     ccversion=''\, gccversion='3\.4\.4 \[FreeBSD\] 20050518'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='gcc'\, ldflags ='\-Wl\,\-E  \-L/usr/local/lib'
>     libpth=/usr/lib /usr/local/lib
>     libs=\-lm \-lcrypt \-lutil \-lc
>     perllibs=\-lm \-lcrypt \-lutil \-lc
>     libc=\, so=so\, useshrplib=false\, libperl=libperl\.a
>     gnulibc\_version=''
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags=' '
>     cccdlflags='\-DPIC \-fPIC'\, lddlflags='\-shared  \-L/usr/local/lib'
> 
> Locally applied patches&#8203;:
>     
> 
> \-\-\-
> @&#8203;INC for perl 5\.9\.4&#8203;:
>     lib
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4/i386\-freebsd
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4/i386\-freebsd
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl
>     \.
> 
> \-\-\-
> Environment for perl 5\.9\.4&#8203;:
>     HOME=/home/nicholas
>     LANG \(unset\)
>     LANGUAGE \(unset\)
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>     PATH=/home/nicholas/bin&#8203;:/sbin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/usr/bin&#8203;:/usr/games&#8203;:/usr/local/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11R6/bin&#8203;:/home/nicholas/bin&#8203;:/usr/local/sbin&#8203;:/sbin&#8203;:/usr/sbin&#8203;:\.
>     PERL\_BADLANG \(unset\)
>     SHELL=/usr/local/bin/bash
> 
> That's all I have\. It seems that if the Makefile\.PL for Math&#8203;::BigInt&#8203;::FastCalc
> is run at \*exactly\* the wrong time\, somewhere in the middle of List&#8203;::Util
> being copied into lib\, then it fails\.
> 
> But I've been unable to replicate this by doing nasty things like seeding
> lib with all generated files truncated to zero length \(so that require will
> fail\)
> 
> So I'm stuck
> 
> Nicholas Clark
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=low
> \-\-\-
> Site configuration information for perl 5\.9\.4&#8203;:
> 
> Configured by nicholas at Thu Feb  2 00&#8203;:04&#8203;:15 CET 2006\.
> 
> Summary of my perl5 \(revision 5 version 9 subversion 4\) configuration&#8203;:
>   Platform&#8203;:
>     osname=freebsd\, osvers=6\.0\-stable\, archname=i386\-freebsd
>     uname='freebsd profane\.mongueurs\.net 6\.0\-stable freebsd 6\.0\-stable \#4&#8203;: wed jan 25 15&#8203;:42&#8203;:18 cet 2006 root@&#8203;profane\.mongueurs\.net&#8203;:usrobjusrsrcsysprofane i386 '
>     config\_args='\-Dusedevel=y \-Dcc=ccache gcc \-Dld=gcc \-Ubincompat5005 \-Uinstallusrbinperl \-Dcf\_email=nick@&#8203;ccl4\.org \-Dperladmin=nick@&#8203;ccl4\.org \-Dinc\_version\_list=  \-Dinc\_version\_list\_init=0 \-Doptimize=\-Os \-Uusethreads \-Uuse64bitint \-Uusemymalloc \-Duseperlio \-Dprefix=~/Sandpit/snap5\.9\.x\-27046 \-Dinstallman1dir=none \-Dinstallman3dir=none \-de'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     useithreads=undef\, usemultiplicity=undef
>     useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
>     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='ccache gcc'\, ccflags ='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe \-Wdeclaration\-after\-statement \-I/usr/local/include'\,
>     optimize='\-Os'\,
>     cppflags='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe \-Wdeclaration\-after\-statement \-I/usr/local/include'
>     ccversion=''\, gccversion='3\.4\.4 \[FreeBSD\] 20050518'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='gcc'\, ldflags ='\-Wl\,\-E  \-L/usr/local/lib'
>     libpth=/usr/lib /usr/local/lib
>     libs=\-lm \-lcrypt \-lutil \-lc
>     perllibs=\-lm \-lcrypt \-lutil \-lc
>     libc=\, so=so\, useshrplib=false\, libperl=libperl\.a
>     gnulibc\_version=''
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags=' '
>     cccdlflags='\-DPIC \-fPIC'\, lddlflags='\-shared  \-L/usr/local/lib'
> 
> Locally applied patches&#8203;:
>     
> 
> \-\-\-
> @&#8203;INC for perl 5\.9\.4&#8203;:
>     lib
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4/i386\-freebsd
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4/i386\-freebsd
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4
>     /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl
>     \.
> 
> \-\-\-
> Environment for perl 5\.9\.4&#8203;:
>     HOME=/home/nicholas
>     LANG \(unset\)
>     LANGUAGE \(unset\)
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>     PATH=/home/nicholas/bin&#8203;:/sbin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/usr/bin&#8203;:/usr/games&#8203;:/usr/local/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11R6/bin&#8203;:/home/nicholas/bin&#8203;:/usr/local/sbin&#8203;:/sbin&#8203;:/usr/sbin&#8203;:\.
>     PERL\_BADLANG \(unset\)
>     SHELL=/usr/local/bin/bash

Actually\, I'm thinking Encode is the problem here\.  
Math&#8203;::BigInt&#8203;::FastCalc doesn't appear to use overload\, while Encode 
appears to have been just completing then\, and it does use overload\.pm 
while running\.  The next best suspect is Data&#8203;::Dumper\, but I've always 
found that Encode takes the longest time to build by far in ext/\.

Steve Peters
steve@&#8203;fisharerojo\.org

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @smpeters

> \[stmpeters \- Wed Feb 01 17&#8203;:20&#8203;:33 2006\]&#8203;:
> 
> On Wed\, Feb 01\, 2006 at 04&#8203;:33&#8203;:08PM \-0800\, Nicholas Clark wrote&#8203;:
> > \# New Ticket Created by  Nicholas Clark
> > \# Please include the string&#8203;:  \[perl \#38404\]
> > \# in the subject line of all future correspondence about this issue\.
> > \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=38404 >
> >
> >
> >
> > This is a bug report for perl from nick@&#8203;ccl4\.org\,
> > generated with the help of perlbug 1\.35 running under perl 5\.9\.4\.
> >
> >
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please enter your report here\]
> >
> > The perl make process still has at least 1 race condition\. I've had
>    this
> > exact failure before\, but didn't grab the output at the time\.
> >
> >
> > Writing Makefile for MIME&#8203;::Base64
> >         Making Math&#8203;::BigInt&#8203;::FastCalc \(dynamic\)
> > Can't locate Scalar/Util\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./\.\./\.\./\.\./lib
>    /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4/i386\-freebsd
>    /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/5\.9\.4
>    /home/nicholas/Sandpit/snap5\.9\.x\-
>    27046/lib/perl5/site\_perl/5\.9\.4/i386\-freebsd
>    /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl/5\.9\.4
>    /home/nicholas/Sandpit/snap5\.9\.x\-27046/lib/perl5/site\_perl \.\) at
>    \.\./\.\./\.\./\.\./lib/overload\.pm line 88\.
> > Warning&#8203;: No Makefile\!
> > make&#8203;: don't know how to make config\. Stop
> > make config failed\, continuing anyway\.\.\.
> > make&#8203;: don't know how to make all\. Stop
> > \*\*\* Error code 2
> > Writing compiled form
> > \.\./\.\./\.\./miniperl "\-I\.\./\.\./\.\./lib" "\-I\.\./\.\./\.\./lib"
>    \.\./\.\./\.\./lib/ExtUtils/xsubpp  \-typemap
>    \.\./\.\./\.\./lib/ExtUtils/typemap  Util\.xs > Util\.xsc && mv Util\.xsc
>    Util\.c
> > cp lib/List/Util\.pm \.\./\.\./\.\./lib/List/Util\.pm
> > cp lib/Scalar/Util\.pm \.\./\.\./\.\./lib/Scalar/Util\.pm
> > Running Mkbootstrap for Encode \(\)
> > chmod 644 Encode\.bs
> > cp bin/piconv blib/script/piconv
> > \.\./\.\./miniperl "\-I\.\./\.\./lib" "\-I\.\./\.\./lib" "\-MExtUtils&#8203;::MY" \-e "MY\-
>    >fixin\(shift\)" blib/script/piconv
> > Running Mkbootstrap for List&#8203;::Util \(\)
> > Writing compiled form
> >
> 
> Actually\, Math&#8203;::BigInt&#8203;::FastCalc is an innocent bystander\.
>    overload\.pm
> C\<require>s Scalar&#8203;::Util\, but several modules built first use it\.
> Encode\, for example\, which was being built in another process\.  Here
>    are
> the modules using overload in the ext directory\.
> 
> >
> > That's all I have\. It seems that if the Makefile\.PL for
>    Math&#8203;::BigInt&#8203;::FastCalc
> > is run at \*exactly\* the wrong time\, somewhere in the middle of
>    List&#8203;::Util
> > being copied into lib\, then it fails\.
> >
> > But I've been unable to replicate this by doing nasty things like
>    seeding
> > lib with all generated files truncated to zero length \(so that
>    require will
> > fail\)
> >
> > So I'm stuck
> >
> > Nicholas Clark
> >

Correct me if I'm wrong\, but did change \#27704 fix this one?

Change 27704 by nicholas@&#8203;nicholas\-z1 on 2006/04/03 11&#8203;:32&#8203;:44

&nbsp;    Two modules in ext/ have dependencies that are post\-miniperl\.
&nbsp;    Encode this information in the Makefile for now\.

**Comment by p5pRT** [comment]
### From @nwc10

On Wed\, Apr 19\, 2006 at 12&#8203;:50&#8203;:25PM \-0700\, Steve Peters via RT wrote&#8203;:
> Correct me if I'm wrong\, but did change \#27704 fix this one?
> 
> Change 27704 by nicholas@&#8203;nicholas\-z1 on 2006/04/03 11&#8203;:32&#8203;:44
> 
>     Two modules in ext/ have dependencies that are post\-miniperl\.
>     Encode this information in the Makefile for now\.

So as you're right\, Warnock should apply? :\-\)

Nicholas Clark

PS Technically there are still some dependencies that aren't quite there\.
&nbsp;   IIRC If you haven't run make depend\, then make depend will run in parallel
&nbsp;   with the make\. I'm not sure if that's actually a problem\.
&nbsp;   Unless the build gets into x2p ahead of make depend finishing\. You can
&nbsp;   foul up there\. That one probably needs fixing\, but it's a separate ticket\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
